tensorflow/core/kernels/sampled_addmm_op.cc

```diff
@@ -0,0 +1,137 @@
+#define EIGEN_USE_THREADS
+
+#include "tensorflow/core/framework/op_kernel.h"
+#include "tensorflow/core/kernels/sampled_addmm_op.h"
+#include "tensorflow/core/util/work_sharder.h"
+
+namespace tensorflow {
+
+typedef Eigen::ThreadPoolDevice CPUDevice;
+typedef Eigen::GpuDevice GPUDevice;
+
+namespace functor {
+
+template <typename T>
+struct SampledADDMMFunctor<CPUDevice, T> {
+  static Status Compute(OpKernelContext* ctx, const Tensor& indices_t,
+                     const Tensor& values_t, const Tensor& dense_shape_t,
+                     const Tensor& mat1, const Tensor& mat2, 
+                     const int32_t batch_size, const T beta_, const T alpha_,
+                     const int32_t mat1_num_rows, const int32_t mat1_num_cols,
+                     const int32_t mat2_num_rows, const int32_t mat2_num_cols,
+                     const int32_t mat_num_batches, Tensor* out) {
+    const T* mat1_ptr = mat1.flat<T>().data();
+    const T* mat2_ptr = mat2.flat<T>().data();
+    const T* values_ptr = values_t.flat<T>().data();
+    const int32_t* indices_ptr = indices_t.flat<int32_t>().data();
+    const int32_t* ds_ptr = dense_shape_t.flat<int32_t>().data();
+
+    int32_t ds_num_rows = ds_ptr[0];
+    int32_t ds_num_cols = ds_ptr[1];
```

Comment:
Getting a segmentation fault when I perform this same operation from the gpu implementation of this function